### PR TITLE
Print flow type annotation comments as comments

### PR DESCRIPTION
**Pull request by duailibe**
Flow AST does not expose that a type annotation `a /*: string*/` comes from a comment and so prettier currently prints them as if they were originally inserted as `a: string`. We're trying to identify if a type annotation was made as a comment and print it like so.

[Preview here](https://deploy-preview-3449--prettier.netlify.com/playground/#N4Igxg9gdgLgprEAuEAbOMAEAzCFMD0AVJqWZkuQM4wBOAllAOYDaAuuaUQZgLyYsAOiAAWcVKgjC2AbhAAaEBAAOMetCrJQAQ1q0IAdwAKuhJpTaAbhHoATBSG01k2baipxFAI1rawAawwAZWU-RiZkOgBXTxBbCDAXNw9FRg9aGCNfJgBbbST3WIArKgAPACFfAODtHLgAGUY4ApSlKJhldoAmFtjQ2nTkEGxJAwdlBlgAdTsYEWQADgAGRQmIDynfZSGJuHTLZsVaOABHKPpjrO1c-KRXQsUPHPpI2hjH8PQARSiIeF7FDBtF4ZrY5sguoDfPRUOEAMIQHJ5IZQaCHEBRDwAFWB5nuHgAvgSgA)

### CODE REVIEW

**Comment by duailibe** [comment]
cc @vjeux

**Comment by duailibe** [comment]
I'll update the PR soon-ish to support the ["comment type include"](https://flow.org/en/docs/types/comments/#toc-comment-type-include) syntax as well.

**Comment by vjeux** [comment]
This is awesome! Can’t wait to see it ship :)

**Comment by duailibe** [comment]
I pushed another commit to handle `/* *-/` comments inside the type annotation comments and have decided to handle `/*:: */` include comments in another PR :)

File: tests/comments/__snapshots__/jsfmt.spec.js.snap [hunk 1]
```diff
@@ -339,31 +339,24 @@ KEYPAD_NUMBERS.map((
 
 function f /* f */() {}
 function f(/* args */) {}
-function f() /* returns */ {
-}
-function f /* f */(/* args */) /* returns */ {
-}
+function f() /* returns */ {}
+function f /* f */(/* args */) /* returns */ {}
```
Comment (duailibe) [inline tests/comments/__snapshots__/jsfmt.spec.js.snap]: This was necessary to align babylon and flow.. it was printing:

```js
// flow
function A() /*: T */ {}

// babylon
function A() /*: T */ {
}

**Comment by duailibe** [comment]
This PR is ready for review now :-)

File: tests/comments/__snapshots__/jsfmt.spec.js.snap (hunk 1 above)
Comment (vjeux) [reply tests/comments/__snapshots__/jsfmt.spec.js.snap]: The fact that there was a newline here in the first place was just me being lazy :)

**Comment by duailibe** [comment]
@vjeux I was just taking your feedback about formatting the comment. Can you take another quick look?

**Comment by duailibe** [comment]
Thanks for the reviews!

**Pull request merged**